internal/telemetry/telemetry_test.go

```diff
@@ -36,3 +41,116 @@ func TestProductChange(t *testing.T) {
 	var productsPayload *Products = secondBody.Payload.(*Products)
 	assert.Equal(t, productsPayload.Profiler.Enabled, true)
 }
+
+// mockServer initializes a server that expects a strict amount of telemetry events. It saves these
+// events in a slice until the expected number of events is reached.
+// the `telemetry` argument accepts a function that should run some sequence of telemetry events.
+// the `expectedHits` argument specifies the number of telemetry events the server should expect.
+func mockServer(ctx context.Context, t *testing.T, expectedHits int, telemetry func(), exclude ...RequestType) (waitForEvents func() []string, cleanup func()) {
```

Comment:
```suggestion
func mockServer(ctx context.Context, t *testing.T, expectedHits int, genTelemetry func(), exclude ...RequestType) (waitForEvents func() []string, cleanup func()) {
```
Can we also edit the comment above, maybe something like:
`the provided genTelemetry function should generate the expected telemetry events via calls to the global client`